actions/login.php

```diff
@@ -38,23 +37,20 @@
 
 $result = elgg_authenticate($username, $password);
 if ($result !== true) {
-	register_error($result);
-	forward(REFERER);
+	return elgg_error_response($result, REFERRER, ELGG_HTTP_FORBIDDEN);
```

Comment:
Probably not. Let's default `elgg_error_response()` to `200`. I think that would be more BC as well. Once this is in, I want to allow throwing HTTP exceptions from views and actions, since we now catch those and serve a proper response.